### PR TITLE
Syncing added very messily. Will need some refactoring for the code

### LINKED ISSUE #8: Add a way to manipulate the "requires" and "goals" fields.
- [ ] Goals field should be an array
- [ ] Add a way to manipulate them without manually writing an array

Some input field with a button and ul that lists the data in that field.

### LINKED ISSUE #16: Editor crashes when trying to add new items
it seems to have crashed adding an item, looks like its adding the incorrect objectID instead of the real one
kind: '[ObjectId]',
value: "[ 'IR-2d8157c3-8fc0-414e' ]",
path: 'requires.0',
reason: CastError: Cast to ObjectId failed for value "IR-2d8157c3-8fc0-414e" (type string) at path "requires"

### LINKED ISSUE #20: Editor crashes when trying to export tree with duplicate nodes
Having two nodes with the same title and level crashes the editor.

### PR DESCRIPTION
**Pull request by OmerSabic**
Finally, you can push the changes to the API.
Hopefully no bugs but they are usually everywhere.
Exporting works by pressing the "Export JSON" button (should probably change that), inputting the API info and pressing "Update Tree".
Currently works on the "task" API branch.

### CODE REVIEW

**Review by seanjhardy — APPROVED** [review]

**Review by OmerSabic — COMMENTED** [review]
Wrongly named this commit @OmerSabic
[added checking for duplicates before exporting](https://github.com/Project-Skill-Tree/Skill-Tree-Editor/pull/19/commits/2ac314c96b13e094951550049b694045a91adf46), should be "added editor for arrays"

File: example-tree.json [hunk 1]
```diff
@@ -1,65 +1,74 @@
...
-        "title": "Mindfulness I",
-        "level": 1,
-        "goal": "Mindfulness 2mins",
-        "frequency": "DAILY",
-        "timelimit": "1x Week",
-        "xp": 50,
+        "title": "babble",
+        "level": 2,
+        "goal": [
+            "annihalate infants",
```
Comment (swatarianess) [inline example-tree.json]: change that to something else like "example goal" 😆

File: example-tree.json [hunk 2]
```diff
@@ -1,65 +1,74 @@
...
-        "requires": [],
-        "children": []
+        "name": "item test",
+        "link": "omersabic.xyz",
+        "emoji": "😋",
+        "requires": [
+            "62968af5af70a3169a1a6fc9"
+        ],
+        "type": "items",
+        "id": "62968c920e15c4cb06a8283b"
```
Comment (swatarianess) [inline example-tree.json]: Probably don't want to use "real" id's, if these are real. Unless really required.

Same goes for line 69, 61, 58, etc.

**Review by swatarianess — CHANGES_REQUESTED** [review]

File: example-tree.json (hunk 1 above)
Comment (OmerSabic) [reply example-tree.json]: Sorry about that one, was listening to infant annihilator and couldn't come up with a better example. Will do

File: example-tree.json (hunk 2 above)
Comment (OmerSabic) [reply example-tree.json]: Sounds good, bit of an oversight but will need to change that. Thanks

**Review by swatarianess — APPROVED** [review]

**Pull request merged**